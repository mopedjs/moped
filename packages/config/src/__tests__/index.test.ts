import {resolve} from 'path';
import {loadConfigFile} from '../';

function testValid(filename: string, expectedFiles: string[] = []) {
  expectedFiles = expectedFiles.map(filename =>
    resolve(__dirname + '/valid-configs/' + filename),
  );
  test('valid - ' + filename, () => {
    expect(
      loadConfigFile(__dirname + '/valid-configs/' + filename, {
        accessSync: (filename: string) => {
          if (expectedFiles.indexOf(filename) === -1) {
            throw new Error('Unexpected file ' + filename);
          }
        },
        shortenPathsForTests: true,
      }),
    ).toMatchSnapshot();
  });
}
function testInvalid(filename: string, expectedFiles: string[] = []) {
  expectedFiles = expectedFiles.map(filename =>
    resolve(__dirname + '/invalid-configs/' + filename),
  );
  test('invalid - ' + filename, () => {
    expect(() =>
      loadConfigFile(__dirname + '/invalid-configs/' + filename, {
        accessSync: (filename: string) => {
          if (expectedFiles.indexOf(filename) === -1) {
            throw new Error('Unexpected file ' + filename);
          }
        },
        shortenPathsForTests: true,
      }),
    ).toThrowErrorMatchingSnapshot();
  });
}

testValid('basic.json', ['src/server.tsx']);
testValid('monorepo.json', [
  'src/server.tsx',
  'src/client.tsx',
  'src/index.html',
  'src/some-backend-service.ts',
  'src/public',
  'src/db-migrations',
  'src/db-overrides/index.ts',
]);
testInvalid('additional-properties.json');
testInvalid('empty-monorepo.json');
testInvalid('incorrect-type.json');
