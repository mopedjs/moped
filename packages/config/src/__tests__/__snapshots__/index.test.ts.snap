// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid - additional-properties.json 1`] = `"additional-properties.json should NOT have additional properties"`;

exports[`invalid - empty-monorepo.json 1`] = `"empty-monorepo.json .monorepo should be equal to constant"`;

exports[`invalid - incorrect-type.json 1`] = `"incorrect-type.json .disableClient should be boolean"`;

exports[`valid - basic.json 1`] = `
Object {
  "config": Object {
    "clientEntryPoint": null,
    "dbMigrations": null,
    "dbOverrides": null,
    "dbSchema": "/Users/fplindesay/Documents/GitHub/moped/packages/config/src/__tests__/valid-configs/src/db-schema",
    "dbURL": null,
    "htmlTemplate": null,
    "port": null,
    "publicDirectory": null,
    "serverEntryPointDev": "src/server.tsx",
    "serverEntryPointProd": "src/server.tsx",
  },
  "monorepo": false,
}
`;

exports[`valid - monorepo.json 1`] = `
Object {
  "configs": Object {
    "app": Object {
      "clientEntryPoint": "src/client.tsx",
      "dbMigrations": "src/db-migrations",
      "dbOverrides": "src/db-overrides/index.ts",
      "dbSchema": "/Users/fplindesay/Documents/GitHub/moped/packages/config/src/__tests__/valid-configs/src/db-schema",
      "dbURL": null,
      "htmlTemplate": "src/index.html",
      "port": null,
      "publicDirectory": "src/public",
      "serverEntryPointDev": "src/server.tsx",
      "serverEntryPointProd": "src/server.tsx",
    },
    "some-backend-service": Object {
      "clientEntryPoint": null,
      "dbMigrations": "src/db-migrations",
      "dbOverrides": "src/db-overrides/index.ts",
      "dbSchema": "/Users/fplindesay/Documents/GitHub/moped/packages/config/src/__tests__/valid-configs/src/db-schema",
      "dbURL": null,
      "htmlTemplate": null,
      "port": null,
      "publicDirectory": null,
      "serverEntryPointDev": "src/some-backend-service.ts",
      "serverEntryPointProd": "src/some-backend-service.ts",
    },
  },
  "monorepo": true,
}
`;
